atomic_reactor/inner.py

```diff
@@ -643,77 +638,26 @@ def throw_canceled_build_exception(self, *args, **kwargs):
+
+        When argument ``keep_plugins_running`` is set, the specified plugins
+        are all ensured to be executed and the SIGTERM signal is ignored.
         """
         print_version_of_tools()
-
-        exception_being_handled = False
-        # Make sure exit_runner is defined for finally block
-        exit_runner = None
         try:
             self.fs_watcher.start()
-            signal.signal(signal.SIGTERM, self.throw_canceled_build_exception)
-            prebuild_runner = PreBuildPluginsRunner(self, self.plugins.prebuild,
-                                                    plugin_files=self.plugin_files)
-            prepublish_runner = PrePublishPluginsRunner(self, self.plugins.prepublish,
-                                                        plugin_files=self.plugin_files)
-            postbuild_runner = PostBuildPluginsRunner(self, self.plugins.postbuild,
-                                                      plugin_files=self.plugin_files)
-            # time to run pre-build plugins, so they can access cloned repo
-            logger.info("running pre-build plugins")
-            try:
-                prebuild_runner.run()
-            except PluginFailedException as ex:
-                logger.error("one or more prebuild plugins failed: %s", ex)
-                raise
-
-            # we are delaying initialization, because prebuild plugin reactor_config
-            # might change build method
-            buildstep_runner = BuildStepPluginsRunner(self, self.plugins.buildstep,
-                                                      plugin_files=self.plugin_files)
-
-            logger.info("running buildstep plugins")
-            try:
-                buildstep_runner.run()
-            except PluginFailedException as ex:
-                logger.error('buildstep plugin failed: %s', ex)
-                raise
-
-            # run prepublish plugins
-            try:
-                prepublish_runner.run()
-            except PluginFailedException as ex:
-                logger.error("one or more prepublish plugins failed: %s", ex)
-                raise
-
-            try:
-                postbuild_runner.run()
-            except PluginFailedException as ex:
-                logger.error("one or more postbuild plugins failed: %s", ex)
-                raise
-        except Exception as ex:
-            logger.debug("caught exception (%s) so running exit plugins", exception_message(ex))
-            exception_being_handled = True
-            raise
+            if not self.keep_plugins_running:
+                signal.signal(signal.SIGTERM, self.throw_canceled_build_exception)
+            runner = PluginsRunner(self,
+                                   self.plugins_conf,
+                                   self.plugin_files,
+                                   self.keep_plugins_running,
+                                   plugins_results=self.data.plugins_results)
+            runner.run()
```

Comment:
Shouldn't we ignore SIGTERM explicitly, via `signal.signal(signal.SIGTERM, lambda *args: None)`?

Reply:
Or maybe better, `signal.signal(signal.SIGTERM, signal.SIG_IGN)`

Reply:
PTAL